File: NERP.RND.Web/ClientApp/src/app/entities/models/baseNetworkEntity.ts
import { DbBaseEntity } from '../../common/models/dbBaseEntity';
import { CustomField } from '../../common/models/customField';
import { FileAttachment } from '../../common/models/FileAttachment';
import { Supplier } from '../../common/models/supplier';

export class BaseNetworkEntity extends DbBaseEntity {

    name?: string;
    icon?: string;
    type?: string;
    brand?: string;
    model?: string;
    zoomLevel?: number;

    code?: string;
    regionId?: string;
    parentId?: string;
    parentEntityType?: string;
    constructionStage?: string;
    activationStage?: string;
    accessibility?: string;
    hierarchyType?: string;
    customFields?: CustomField[] = new Array<CustomField>();
    photos?: FileAttachment[] = new Array<FileAttachment>();
    attachments?: FileAttachment[] = new Array<FileAttachment>();
    suppliers?: Supplier[] = new Array<Supplier>();

    constructor(id?: string, code?: string) {
        super();
        this.id = id;
        this.code = code;
    }
}

File: NERP.NetworkEntities/Models/BaseNetworkEntity.cs
﻿using Common.DataAccess.Mongo;
using NERP.Common.Models;
using System.Collections.Generic;

namespace NERP.NetworkEntities.Models
{
    public class BaseNetworkEntity : DbBaseEntity
    {
        public string EntityType { get; set; }
        public string Name { get; set; }
        public string Code { get; set; }
        public string Icon { get; set; }
        public string RegionId { get; set; }
        public string ParentId { get; set; }
        public string ParentEntityType { get; set; }
        public string ConstructionStage { get; set; }
        public string ActivationStage { get; set; }
        public string Accessibility { get; set; }
        public string HierarchyType { get; set; }
        public string Type { get; set; }
        public string Brand { get; set; }
        public string Model { get; set; }
        public int ZoomLevel { get; set; }
        public List<CustomField> CustomFields { get; set; }
        public List<FileAttachment> Photos { get; set; }
        public List<FileAttachment> Attachments { get; set; }
        public List<Supplier> Suppliers { get; set; }
    }
}
